reference/sql/characterset-and-collation.md

```diff
@@ -324,4 +280,100 @@ SET collation_connection = @@collation_database;
 
 如果不希望报错，可以通过 `set @@tidb_skip_utf8_check=1;` 跳过字符检查。
 
+## 排序规则支持
+
+Collation 的语法支持和语义支持受到配置项 [new_collation_enable](/reference/configuration/tidb-server/configuration-file.md#new_collations_enabled_on_first_bootstrap) 的影响。 这里我们区分语法支持和语义支持，语法支持是指 TiDB 能够解析和设置 collation；而语义支持是指 TiDB 能够在字符串比较时正确地使用 collation。
+
+在 4.0 版本之前，TiDB 只提供了旧的 collation 框架，能够在语法上支持的绝大部分 MySQL collation，但语义上所有的 collation 都当成 binary collation。
+
+4.0 版本中，TiDB 增加了新的 collation 框架用于在语义上支持不同的 collation，保证字符串比较时严格遵循对应的 collation，详情请见下。
+
+### 旧框架下的 Collation 支持
+
+在 4.0 版本之前，TiDB 中可以指定大部分 MySQl 中的 Collation，并把它们按照默认 Collation 处理，即以编码字节序为字符定序。同时，并未像 MySQL 一样，在比较前按照 Collation 的 `PADDING` 属性将字符补齐空格。因此，会造成以下的行为区别：
+
+{{< copyable "sql" >}}
+
+```sql
+tidb> create table t(a varchar(20) charset utf8mb4 collate utf8mb4_general_ci primary key);
+Query OK, 0 rows affected
+tidb> insert into t values ('A');
+Query OK, 1 row affected
+tidb> insert into t values ('a');
+Query OK, 1 row affected # MySQL 中，由于 utf8mb4_general_ci 大小写不敏感，报错 Duplicate entry 'a'.
+tidb> insert into t1 values ('a ');
+Query OK, 1 row affected # MySQL 中，由于补齐空格比较，报错 Duplicate entry 'a '
```

Comment:
```suggestion
Query OK, 1 row affected # MySQL 中，由于补齐空格比较，报错 Duplicate entry 'a'。
```

Reply:
Space here is necessary. It should be `'a '。`.